src/olm.cpp

```diff
@@ -588,34 +597,45 @@ size_t olm_create_outbound_session(
     OlmSession * session,
     OlmAccount const * account,
     void const * their_identity_key, size_t their_identity_key_length,
+    void const * their_signing_key, size_t their_signing_key_length,
     void const * their_pre_key, size_t their_pre_key_length,
+    void const * their_pre_key_signature, size_t their_pre_key_signature_length,
     void const * their_one_time_key, size_t their_one_time_key_length,
     void * random, size_t random_length
 ) {
     std::uint8_t const * id_key = from_c(their_identity_key);
+    std::uint8_t const * s_key = from_c(their_signing_key);
     std::uint8_t const * ot_key = from_c(their_one_time_key);
     std::uint8_t const * p_key = from_c(their_pre_key);
+    std::uint8_t const * p_sign = from_c(their_pre_key_signature);
     std::size_t id_key_length = their_identity_key_length;
+    std::size_t s_key_length = their_signing_key_length;
     std::size_t ot_key_length = their_one_time_key_length;
     std::size_t p_key_length = their_pre_key_length;
+    std::size_t p_sign_length = their_pre_key_signature_length;
 
     if (olm::decode_base64_length(id_key_length) != CURVE25519_KEY_LENGTH
+            || olm::decode_base64_length(s_key_length) != ED25519_PUBLIC_KEY_LENGTH
             || olm::decode_base64_length(ot_key_length) != CURVE25519_KEY_LENGTH
             || olm::decode_base64_length(p_key_length) != CURVE25519_KEY_LENGTH
     ) {
         from_c(session)->last_error = OlmErrorCode::OLM_INVALID_BASE64;
         return std::size_t(-1);
     }
     _olm_curve25519_public_key identity_key;
+    _olm_ed25519_public_key signing_key;
     _olm_curve25519_public_key pre_key;
     _olm_curve25519_public_key one_time_key;
+    uint8_t raw_pre_key_signature[ED25519_SIGNATURE_LENGTH];
 
     olm::decode_base64(id_key, id_key_length, identity_key.public_key);
+    olm::decode_base64(s_key, s_key_length, signing_key.public_key);
     olm::decode_base64(ot_key, ot_key_length, one_time_key.public_key);
     olm::decode_base64(p_key, p_key_length, pre_key.public_key);
+    olm::decode_base64(p_sign, p_sign_length, raw_pre_key_signature);
```

Comment:
It doesn't appear that we're doing any bounds checking on the prekey signature. If a malicious user sends a extra-long prekey signature, would it be possible that the bounds of `raw_pre_key_signature` will overflow, leading to a potential buffer overflow attack?